src/sentry/integrations/slack/event_endpoint.py

```diff
@@ -165,46 +167,50 @@ def on_message(self, request, integration, token, data):
         return self.respond()
 
     def on_link_shared(self, request, integration, token, data):
-        parsed_issues = defaultdict(dict)
-        event_id_by_url = {}
-        for item in data["links"]:
-            event_type, instance_id, event_id = self._parse_url(item["url"])
-            if not instance_id:
-                continue
-            # note that because we store the url by the issue,
-            # we will only unfurl one link per issue even if there are
-            # multiple links to different events
-            parsed_issues[event_type][instance_id] = item["url"]
-            event_id_by_url[item["url"]] = event_id
-
-        if not parsed_issues:
-            return
-
-        results = {}
-        for event_type, instance_map in parsed_issues.items():
-            results.update(
-                self.event_handlers[event_type](integration, instance_map, event_id_by_url)
-            )
-
-        if not results:
-            return
-
-        access_token = self._get_access_token(integration)
-
-        payload = {
-            "token": access_token,
-            "channel": data["channel"],
-            "ts": data["message_ts"],
-            "unfurls": json.dumps(results),
-        }
+        with sentry_sdk.start_transaction(
+            op=u"slack.link_shared", name=u"SlackLinkShared", sampled=1.0
+        ) as span:
+            parsed_issues = defaultdict(dict)
+            event_id_by_url = {}
+            for item in data["links"]:
+                span.set_tag("link", parse_link(item["url"]))
```

Comment:
@ceorourke I would hate for this to break because `parse_link` fails. Maybe we should wrap this in a try/catch to make sure there isn't some edge case that blows up `parse_link`.